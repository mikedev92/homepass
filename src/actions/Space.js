// @flow
import { type Action } from './types';

export function setFilter(key: string, value: boolean): Action { // eslint-disable-line
  return {
    type: 'SET_FILTER',
    key,
    value,
  };
}
