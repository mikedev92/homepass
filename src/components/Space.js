// @flow
import React from 'react';
import { TouchableOpacity } from 'react-native';
import { Col, ListItem, H1, Text, Container, Left, Button, Icon, Thumbnail, Body, Right, View } from 'native-base';
import { graphql } from 'react-apollo';
import gql from 'graphql-tag';
import { compose } from 'recompose';
import { get, range } from 'lodash';
import { graphqlLoader, CalendarEventView, HorizontalScrollView, ContactSummaryView, Feed, CollapsibleHeader } from './common';
import { variables } from '../theme';

type Props = {
	navigation: Navigation,
	data: ApolloData<Space>,
	joinSpace: () => void
};

function getJoinSpaceButtonText(space: Space) {
	if (space.viewerMemberRequest) {
		return 'Join Requested';
	}
	return space.visibility === 'OPEN' ? 'Join' : 'Request To Join';
}
function SpaceView({ data, navigation, joinSpace }: Props) {
	const space = data.node;
	if (!space) {
		return <Container />;
	}
	return (
		<CollapsibleHeader imageSource={{ uri: space.imageUrl }} title={space.title} onBackPress={() => navigation.goBack()}>
			<ListItem noBorder>
				<Col>
					<H1>
						{get(data, 'node.title')}
					</H1>
					<Text caption darkGray numberOfLines={2} ellipsizeMode="tail">
						{get(data, 'node.resources[0].spaceResourceDescription')}
					</Text>
				</Col>
			</ListItem>
			<ListItem noBorder>
				<Body style={{ flexDirection: 'row', overflow: 'hidden' }}>
					{((space.members && space.members.edges) || []).map(
						(edge: SpaceMemberEdge) =>
							edge.node &&
							edge.node.user &&
							edge.node.user.imageUrl &&
							<TouchableOpacity key={get(edge, 'node.user.id')} onPress={() => navigation.navigate('MemberView', { id: get(edge, 'node.user.id') })}>
								<Thumbnail style={{ margin: 5 }} tiny source={{ uri: get(edge, 'node.user.imageUrl') }} />
							</TouchableOpacity>
					)}
				</Body>
				{space.viewerMember &&
					<Right>
						<Button onPress={() => navigation.navigate('SpaceMemberships', { id: space.id })} transparent noPadding>
							<Icon style={{ fontSize: 24 }} name="add-circle" />
						</Button>
					</Right>}
			</ListItem>
			{!space.viewerMember &&
				<Button onPress={joinSpace} style={{ marginHorizontal: variables.contentPadding }} primary={!space.viewerMemberRequest} disabled={Boolean(space.viewerMemberRequest)} bordered={Boolean(space.viewerMemberRequest)} block>
					<Text>
						{getJoinSpaceButtonText(space)}
					</Text>
				</Button>}
			{space.viewerMember &&
				<View>
					<ListItem itemDivider>
						<Body>
							<Text title>Events</Text>
						</Body>
						<Right>
							<Button transparent noPadding onPress={() => navigation.navigate('SpaceEvents')}>
								<Text>See all</Text>
							</Button>
						</Right>
					</ListItem>
					<HorizontalScrollView>
						{get(data, 'node.calendarEvents.edges', []).map((edge: CalendarEventEdge) => edge.node && <CalendarEventView key={edge.node.id} event={edge.node} />)}
					</HorizontalScrollView>
					<ListItem itemDivider>
						<Body>
							<Text title>Top Contacts</Text>
						</Body>
						<Right>
							<Button transparent noPadding onPress={() => navigation.navigate('SpaceContacts')}>
								<Text>See all</Text>
							</Button>
						</Right>
					</ListItem>
					<HorizontalScrollView>
						{((space.contacts && space.contacts.edges) || []).map((edge: SpaceContactEdge) => edge.node && edge.node.contact && <ContactSummaryView key={edge.node.contact.id} spaceContact={edge.node} />)}
					</HorizontalScrollView>
					<ListItem itemDivider>
						<Body>
							<Text title>Activity</Text>
						</Body>
					</ListItem>
					<ListItem noBorder>
						<Feed>
							{range(10).map(i =>
								<ListItem key={i}>
									<Left>
										<Icon name="arrow-back" />
									</Left>
									<Body>
										<Text>Note added to christina for 27 Elizabeth St</Text>
										<Text status>Today at 9:41AM</Text>
										{i % 4 === 0 ? <Text>Some extra content</Text> : null}
									</Body>
									<Right>
										<Thumbnail source={{ uri: 'https://unsplash.it/200/300' }} />
									</Right>
								</ListItem>
							)}
						</Feed>
					</ListItem>
				</View>}
		</CollapsibleHeader>
	);
}

const SpaceFragment = gql`
  fragment SpaceFragment on Space {
        id
        spaceId
        title
        imageUrl
        visibility
        viewerMember {
          id
        }
        viewerMemberRequest {
          id
        }
        members {
          edges {
            node {
              id
             user {
               id
               imageUrl
             }
            }
          }
        }
        calendarEvents {
          edges {
            node {
              id
              name
              startDate
              endDate
            }
          }
        }
        contacts {
          edges {
            node {
             ...SpaceContactSummaryViewFragment
            }
          }
        }
        resources {
          spaceResourceTitle
          spaceResourceSubtitle
          spaceResourceDescription
        }
  }
  ${ContactSummaryView.fragments.SpaceContact}
`;
const SpaceQuery = gql`
  query SpaceQuery($id: ID!) {
    node(id: $id) {
      ...SpaceFragment
    }
  }
  ${SpaceFragment}
`;

const JoinSpaceMutation = gql`
  mutation joinSpace($input: JoinSpaceInput!) {
    joinSpace(input: $input) {
      space {
        ...SpaceFragment
      }
    }
  }
  ${SpaceFragment}
`;

export default compose(
	graphql(SpaceQuery, {
		options: props => ({ variables: { id: props.navigation.state.params.id } })
	}),
	graphql(JoinSpaceMutation, {
		props: ({ mutate, ownProps }) => ({
			joinSpace: () => mutate({ variables: { input: { spaceId: get(ownProps, 'data.node.spaceId') } } })
		})
	}),
	graphqlLoader
)(SpaceView);
