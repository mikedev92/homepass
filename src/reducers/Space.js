// @flow
import { type Action } from '../actions/types';

export type State = {
  filters: {
    [string]: boolean
  },
}

const initialState = {
  filters: {},
};

export default (state: State = initialState, action: Action): State => {
  switch (action.type) {
    case 'SET_FILTER': {
      return {
        ...state,
        filters: {
          ...state.filters,
          [action.key]: action.value,
        },
      };
    }
    default: return state;
  }
};
