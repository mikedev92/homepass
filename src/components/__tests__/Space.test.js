// @flow
import React from 'react';
import { shallow } from 'enzyme';
import MockDate from 'mockdate';
import Space from '../Space';

const testProps = [
  {
    data: {
      node: {
        id: '123',
        spaceId: '123',
        title: 'Title',
        imageUrl: 'Image',
        visibility: 'OPEN',
        viewerMember: {
          id: '123',
        },
        viewerMemberRequest: null,
        members: {
          edges: [
            {
              node: {
                id: '123',
                user: {
                  id: '123',
                  imageUrl: 'user 1 image',
                },
              },
            },
          ],
        },
        calendarEvents: {
          edges: [
            {
              node: {
                id: '123',
                name: 'Open',
                startDate: new Date(0),
                endDate: new Date(1000000000000000),
              },
            },
          ],
        },
        contacts: {
          edges: [
            {
              node: {
                id: '123',
              },
            },
          ],
        },
        resources: [
          {
            spaceResourceTitle: 'resource title',
            spaceResourceSubtitle: 'resource subtitle',
            spaceResourceDescription: 'resource description',
          },
        ],
      },
    },
  },
  {
    data: {
      node: {
        id: '123',
        spaceId: '123',
        title: 'Title',
        imageUrl: 'Image',
        visibility: 'OPEN',
        viewerMember: null,
        viewerMemberRequest: null,
        members: {
          edges: [],
        },
        calendarEvents: {
          edges: [
            {
              node: {
                id: '123',
                name: 'Open',
                startDate: new Date(0),
                endDate: new Date(1000000000000000),
              },
            },
          ],
        },
        contacts: {
          edges: [
            {
              node: {
                id: '123',
              },
            },
          ],
        },
        resources: [
          {
            spaceResourceTitle: 'resource title',
            spaceResourceSubtitle: 'resource subtitle',
            spaceResourceDescription: 'resource description',
          },
        ],
      },
    },
  },
  {
    data: {
      node: {
        id: '123',
        spaceId: '123',
        title: 'Title',
        imageUrl: 'Image',
        visibility: 'OPEN',
        viewerMember: null,
        viewerMemberRequest: null,
        members: {
          edges: [
            {
              node: {
                id: '123',
                user: {
                  id: '123',
                  imageUrl: 'user 1 image',
                },
              },
            },
          ],
        },
        calendarEvents: {
          edges: [],
        },
        contacts: {
          edges: [
            {
              node: {
                id: '123',
              },
            },
          ],
        },
        resources: [
          {
            spaceResourceTitle: 'resource title',
            spaceResourceSubtitle: 'resource subtitle',
            spaceResourceDescription: 'resource description',
          },
        ],
      },
    },
  },

  {
    data: {
      node: {
        id: '123',
        spaceId: '123',
        title: 'Title',
        imageUrl: 'Image',
        visibility: 'OPEN',
        viewerMember: null,
        viewerMemberRequest: null,
        members: {
          edges: [
            {
              node: {
                id: '123',
                user: {
                  id: '123',
                  imageUrl: 'user 1 image',
                },
              },
            },
          ],
        },
        calendarEvents: {
          edges: [
            {
              node: {
                id: '123',
                name: 'Open',
                startDate: new Date(0),
                endDate: new Date(1000000000000000),
              },
            },
          ],
        },
        contacts: {
          edges: [
            {
              node: {
                id: '123',
              },
            },
          ],
        },
        resources: [
          {
            spaceResourceTitle: 'resource title',
            spaceResourceSubtitle: 'resource subtitle',
            spaceResourceDescription: 'resource description',
          },
        ],
      },
    },
  },
  {
    data: {
      node: {
        id: '123',
        spaceId: '123',
        title: 'Title',
        imageUrl: 'Image',
        visibility: 'PRIVATE',
        viewerMember: null,
        viewerMemberRequest: null,
        members: {
          edges: [
            {
              node: {
                id: '123',
                user: {
                  id: '123',
                  imageUrl: 'user 1 image',
                },
              },
            },
          ],
        },
        calendarEvents: {
          edges: [
            {
              node: {
                id: '123',
                name: 'Open',
                startDate: new Date(0),
                endDate: new Date(1000000000000000),
              },
            },
          ],
        },
        contacts: {
          edges: [
            {
              node: {
                id: '123',
              },
            },
          ],
        },
        resources: [
          {
            spaceResourceTitle: 'resource title',
            spaceResourceSubtitle: 'resource subtitle',
            spaceResourceDescription: 'resource description',
          },
        ],
      },
    },
  },
  {
    data: {
      node: {
        id: '123',
        spaceId: '123',
        title: 'Title',
        imageUrl: 'Image',
        visibility: 'OPEN',
        viewerMember: null,
        viewerMemberRequest: { id: '123' },
        members: {
          edges: [
            {
              node: {
                id: '123',
                user: {
                  id: '123',
                  imageUrl: 'user 1 image',
                },
              },
            },
          ],
        },
        calendarEvents: {
          edges: [
            {
              node: {
                id: '123',
                name: 'Open',
                startDate: new Date(0),
                endDate: new Date(1000000000000000),
              },
            },
          ],
        },
        contacts: {
          edges: [
            {
              node: {
                id: '123',
              },
            },
          ],
        },
        resources: [
          {
            spaceResourceTitle: 'resource title',
            spaceResourceSubtitle: 'resource subtitle',
            spaceResourceDescription: 'resource description',
          },
        ],
      },
    },
  },
];

describe('<Space />', () => {
  beforeEach(() => {
    MockDate.set(new Date('1/1/2000'));
  });

  afterEach(() => {
    MockDate.reset();
  });

  it('matches snapshots', () => {
    testProps.forEach((props) => {
      const wrapper = shallow(<Space {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
