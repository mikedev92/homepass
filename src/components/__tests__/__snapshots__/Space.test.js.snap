// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Space /> matches snapshots 1`] = `
<CollapsibleHeader
  imageSource={
    Object {
      "uri": "Image",
    }
  }
  onBackPress={[Function]}
  title="Title"
>
  <Styled(ListItem)
    noBorder={true}
  >
    <ColumnNB>
      <Styled(H1)>
        Title
      </Styled(H1)>
      <Styled(Text)
        caption={true}
        darkGray={true}
        ellipsizeMode="tail"
        numberOfLines={2}
      >
        resource description
      </Styled(Text)>
    </ColumnNB>
  </Styled(ListItem)>
  <Styled(ListItem)
    noBorder={true}
  >
    <Styled(Body)
      style={
        Object {
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        focusedOpacity={0.7}
        onPress={[Function]}
      >
        <Styled(Thumbnail)
          source={
            Object {
              "uri": "user 1 image",
            }
          }
          style={
            Object {
              "margin": 5,
            }
          }
          tiny={true}
        />
      </TouchableOpacity>
    </Styled(Body)>
    <Styled(Right)>
      <Styled(Button)
        noPadding={true}
        onPress={[Function]}
        transparent={true}
      >
        <Styled(Icon)
          name="add-circle"
          style={
            Object {
              "fontSize": 24,
            }
          }
        />
      </Styled(Button)>
    </Styled(Right)>
  </Styled(ListItem)>
  <Styled(ViewNB)>
    <Styled(ListItem)
      itemDivider={true}
    >
      <Styled(Body)>
        <Styled(Text)
          title={true}
        >
          Events
        </Styled(Text)>
      </Styled(Body)>
      <Styled(Right)>
        <Styled(Button)
          noPadding={true}
          onPress={[Function]}
          transparent={true}
        >
          <Styled(Text)>
            See all
          </Styled(Text)>
        </Styled(Button)>
      </Styled(Right)>
    </Styled(ListItem)>
    <HorizontalScrollView>
      <CalendarEventView
        event={
          Object {
            "endDate": +033658-09-27T01:46:40.000Z,
            "id": "123",
            "name": "Open",
            "startDate": 1970-01-01T00:00:00.000Z,
          }
        }
      />
    </HorizontalScrollView>
    <Styled(ListItem)
      itemDivider={true}
    >
      <Styled(Body)>
        <Styled(Text)
          title={true}
        >
          Top Contacts
        </Styled(Text)>
      </Styled(Body)>
      <Styled(Right)>
        <Styled(Button)
          noPadding={true}
          onPress={[Function]}
          transparent={true}
        >
          <Styled(Text)>
            See all
          </Styled(Text)>
        </Styled(Button)>
      </Styled(Right)>
    </Styled(ListItem)>
    <HorizontalScrollView />
    <Styled(ListItem)
      itemDivider={true}
    >
      <Styled(Body)>
        <Styled(Text)
          title={true}
        >
          Activity
        </Styled(Text)>
      </Styled(Body)>
    </Styled(ListItem)>
    <Styled(ListItem)
      noBorder={true}
    >
      <Styled(Feed)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
            <Styled(Text)>
              Some extra content
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
            <Styled(Text)>
              Some extra content
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
            <Styled(Text)>
              Some extra content
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
        <Styled(ListItem)>
          <Styled(Left)>
            <Styled(Icon)
              name="arrow-back"
            />
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Text)>
              Note added to christina for 27 Elizabeth St
            </Styled(Text)>
            <Styled(Text)
              status={true}
            >
              Today at 9:41AM
            </Styled(Text)>
          </Styled(Body)>
          <Styled(Right)>
            <Styled(Thumbnail)
              source={
                Object {
                  "uri": "https://unsplash.it/200/300",
                }
              }
            />
          </Styled(Right)>
        </Styled(ListItem)>
      </Styled(Feed)>
    </Styled(ListItem)>
  </Styled(ViewNB)>
</CollapsibleHeader>
`;

exports[`<Space /> matches snapshots 2`] = `
<CollapsibleHeader
  imageSource={
    Object {
      "uri": "Image",
    }
  }
  onBackPress={[Function]}
  title="Title"
>
  <Styled(ListItem)
    noBorder={true}
  >
    <ColumnNB>
      <Styled(H1)>
        Title
      </Styled(H1)>
      <Styled(Text)
        caption={true}
        darkGray={true}
        ellipsizeMode="tail"
        numberOfLines={2}
      >
        resource description
      </Styled(Text)>
    </ColumnNB>
  </Styled(ListItem)>
  <Styled(ListItem)
    noBorder={true}
  >
    <Styled(Body)
      style={
        Object {
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    />
  </Styled(ListItem)>
  <Styled(Button)
    block={true}
    bordered={false}
    disabled={false}
    primary={true}
    style={
      Object {
        "marginHorizontal": 10,
      }
    }
  >
    <Styled(Text)>
      Join
    </Styled(Text)>
  </Styled(Button)>
</CollapsibleHeader>
`;

exports[`<Space /> matches snapshots 3`] = `
<CollapsibleHeader
  imageSource={
    Object {
      "uri": "Image",
    }
  }
  onBackPress={[Function]}
  title="Title"
>
  <Styled(ListItem)
    noBorder={true}
  >
    <ColumnNB>
      <Styled(H1)>
        Title
      </Styled(H1)>
      <Styled(Text)
        caption={true}
        darkGray={true}
        ellipsizeMode="tail"
        numberOfLines={2}
      >
        resource description
      </Styled(Text)>
    </ColumnNB>
  </Styled(ListItem)>
  <Styled(ListItem)
    noBorder={true}
  >
    <Styled(Body)
      style={
        Object {
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        focusedOpacity={0.7}
        onPress={[Function]}
      >
        <Styled(Thumbnail)
          source={
            Object {
              "uri": "user 1 image",
            }
          }
          style={
            Object {
              "margin": 5,
            }
          }
          tiny={true}
        />
      </TouchableOpacity>
    </Styled(Body)>
  </Styled(ListItem)>
  <Styled(Button)
    block={true}
    bordered={false}
    disabled={false}
    primary={true}
    style={
      Object {
        "marginHorizontal": 10,
      }
    }
  >
    <Styled(Text)>
      Join
    </Styled(Text)>
  </Styled(Button)>
</CollapsibleHeader>
`;

exports[`<Space /> matches snapshots 4`] = `
<CollapsibleHeader
  imageSource={
    Object {
      "uri": "Image",
    }
  }
  onBackPress={[Function]}
  title="Title"
>
  <Styled(ListItem)
    noBorder={true}
  >
    <ColumnNB>
      <Styled(H1)>
        Title
      </Styled(H1)>
      <Styled(Text)
        caption={true}
        darkGray={true}
        ellipsizeMode="tail"
        numberOfLines={2}
      >
        resource description
      </Styled(Text)>
    </ColumnNB>
  </Styled(ListItem)>
  <Styled(ListItem)
    noBorder={true}
  >
    <Styled(Body)
      style={
        Object {
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        focusedOpacity={0.7}
        onPress={[Function]}
      >
        <Styled(Thumbnail)
          source={
            Object {
              "uri": "user 1 image",
            }
          }
          style={
            Object {
              "margin": 5,
            }
          }
          tiny={true}
        />
      </TouchableOpacity>
    </Styled(Body)>
  </Styled(ListItem)>
  <Styled(Button)
    block={true}
    bordered={false}
    disabled={false}
    primary={true}
    style={
      Object {
        "marginHorizontal": 10,
      }
    }
  >
    <Styled(Text)>
      Join
    </Styled(Text)>
  </Styled(Button)>
</CollapsibleHeader>
`;

exports[`<Space /> matches snapshots 5`] = `
<CollapsibleHeader
  imageSource={
    Object {
      "uri": "Image",
    }
  }
  onBackPress={[Function]}
  title="Title"
>
  <Styled(ListItem)
    noBorder={true}
  >
    <ColumnNB>
      <Styled(H1)>
        Title
      </Styled(H1)>
      <Styled(Text)
        caption={true}
        darkGray={true}
        ellipsizeMode="tail"
        numberOfLines={2}
      >
        resource description
      </Styled(Text)>
    </ColumnNB>
  </Styled(ListItem)>
  <Styled(ListItem)
    noBorder={true}
  >
    <Styled(Body)
      style={
        Object {
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        focusedOpacity={0.7}
        onPress={[Function]}
      >
        <Styled(Thumbnail)
          source={
            Object {
              "uri": "user 1 image",
            }
          }
          style={
            Object {
              "margin": 5,
            }
          }
          tiny={true}
        />
      </TouchableOpacity>
    </Styled(Body)>
  </Styled(ListItem)>
  <Styled(Button)
    block={true}
    bordered={false}
    disabled={false}
    primary={true}
    style={
      Object {
        "marginHorizontal": 10,
      }
    }
  >
    <Styled(Text)>
      Request To Join
    </Styled(Text)>
  </Styled(Button)>
</CollapsibleHeader>
`;

exports[`<Space /> matches snapshots 6`] = `
<CollapsibleHeader
  imageSource={
    Object {
      "uri": "Image",
    }
  }
  onBackPress={[Function]}
  title="Title"
>
  <Styled(ListItem)
    noBorder={true}
  >
    <ColumnNB>
      <Styled(H1)>
        Title
      </Styled(H1)>
      <Styled(Text)
        caption={true}
        darkGray={true}
        ellipsizeMode="tail"
        numberOfLines={2}
      >
        resource description
      </Styled(Text)>
    </ColumnNB>
  </Styled(ListItem)>
  <Styled(ListItem)
    noBorder={true}
  >
    <Styled(Body)
      style={
        Object {
          "flexDirection": "row",
          "overflow": "hidden",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        focusedOpacity={0.7}
        onPress={[Function]}
      >
        <Styled(Thumbnail)
          source={
            Object {
              "uri": "user 1 image",
            }
          }
          style={
            Object {
              "margin": 5,
            }
          }
          tiny={true}
        />
      </TouchableOpacity>
    </Styled(Body)>
  </Styled(ListItem)>
  <Styled(Button)
    block={true}
    bordered={true}
    disabled={true}
    primary={false}
    style={
      Object {
        "marginHorizontal": 10,
      }
    }
  >
    <Styled(Text)>
      Join Requested
    </Styled(Text)>
  </Styled(Button)>
</CollapsibleHeader>
`;
